node/cli/src/service.rs

```diff
@@ -110,21 +112,20 @@ construct_service_factory! {
 					service.spawn_task(Box::new(select));
 				}
 
-				let grandpa_key = if service.config.disable_grandpa {
+				let grandpa_session_store = if service.config.disable_grandpa {
 					None
 				} else {
-					service.authority_key::<grandpa_primitives::AuthorityPair>()
+					service.get_session_store::<grandpa_primitives::AuthorityPair>()
 				};
 
 				let config = grandpa::Config {
-					local_key: grandpa_key.map(Arc::new),
 					// FIXME #1578 make this available through chainspec
 					gossip_duration: Duration::from_millis(333),
 					justification_period: 4096,
 					name: Some(service.config.name.clone())
 				};
 
-				match config.local_key {
```

Comment:
Previously when a user specified no key, Substrate would not start a grandpa voter service. Do I understand correctly that if a user does not specify a key and also does not set the `disable-grandpa` flag, substrate is going to start a grandpa voter now?

Reply:
Yes. If you do not want to be a grandpa voter, which is the default, you have to disable it.

Reply:
But the grandpa voter can handle the case where there is no key...

Reply:
@mxinden do we still need the --grandpa-voter flag after this PR?

Reply:
@dvc94ch do I understand correctly, that the only way of only running a grandpa *observer* not a *voter* is through the `disable-grandpa` flag? In that case, no, the `--grandpa-voter` flag is obsolete. Would you mind removing it within this pull request and adjusting the `docker-compose.yml` so the master branch stays consistent? Let me know if you want me to push a commit.

```diff
diff --git a/scripts/sentry-node/docker-compose.yml b/scripts/sentry-node/docker-compose.yml
index 78f8ba36b..661ab9433 100644
--- a/scripts/sentry-node/docker-compose.yml
+++ b/scripts/sentry-node/docker-compose.yml
@@ -93,9 +93,6 @@ services:
       # Not only bind to localhost.
       - "--ws-external"
       - "--rpc-external"
-      # Make sure sentry-a still participates as a grandpa voter to forward
-      # grandpa finality gossip messages.
-      - "--grandpa-voter"
```

Also it might be worth extending the `--help` output a bit:

```diff
diff --git a/core/cli/src/params.rs b/core/cli/src/params.rs
index b22d25a9d..64bdfc692 100644
--- a/core/cli/src/params.rs
+++ b/core/cli/src/params.rs
@@ -297,7 +297,7 @@ pub struct RunCmd {
        #[structopt(long = "validator")]
        pub validator: bool,

-       /// Disable GRANDPA voter
+       /// Disable GRANDPA voter only running a GRANDPA observer
        #[structopt(long = "no-grandpa")]
        pub no_grandpa: bool,
```